typescript.js

```diff
@@ -2,79 +2,60 @@ module.exports = {
-        indent: 'off',
-        '@typescript-eslint/indent': ['error', 4],
-        'keyword-spacing': 'off',
-        '@typescript-eslint/keyword-spacing': 'error',
-        'lines-between-class-members': 'off',
-        '@typescript-eslint/lines-between-class-members': 'error',
-        'no-array-constructor': 'off',
-        '@typescript-eslint/no-array-constructor': 'error',
-        'no-dupe-class-members': 'off',
-        '@typescript-eslint/no-dupe-class-members': 'error',
-        'no-empty-function': 'off',
-        '@typescript-eslint/no-empty-function': ['error', {
-            allow: [
-                'arrowFunctions',
-                'functions',
-                'methods'
-            ]
-        }],
-        'no-extra-semi': 'off',
-        '@typescript-eslint/no-extra-semi': 'error',
-        'no-loop-func': 'off',
-        '@typescript-eslint/no-loop-func': 'error',
-        'no-redeclare': 'off',
-        '@typescript-eslint/no-redeclare': 'error',
-        'no-shadow': 'off',
-        '@typescript-eslint/no-shadow': 'error',
-        'no-unused-expressions': 'off',
-        '@typescript-eslint/no-unused-expressions': 'error',
-        'no-unused-vars': 'off',
-        '@typescript-eslint/no-unused-vars': ['error', { vars: 'all', args: 'after-used', ignoreRestSiblings: true }],
-        'no-use-before-define': 'off',
-        '@typescript-eslint/no-use-before-define': ['error', { functions: false, classes: true, variables: true }],
-        'no-useless-constructor': 'off',
-        '@typescript-eslint/no-useless-constructor': 'error',
-        quotes: 'off',
-        '@typescript-eslint/quotes': ['error', 'single', { avoidEscape: true }],
-        'no-return-await': 'off',
-        '@typescript-eslint/return-await': 'error',
-        semi: 'off',
-        '@typescript-eslint/semi': 'error',
-        'space-before-function-paren': 'off',
-        '@typescript-eslint/space-before-function-paren': ['error', {
-            anonymous: 'always',
-            named: 'never',
-            asyncArrow: 'always'
-        }],
-        'space-infix-ops': 'off',
-        '@typescript-eslint/space-infix-ops': 'error',
+
+        // None
+
+        /*
+            TypeScript rules outside of the recommended configuration
+        */
+        '@typescript-eslint/array-type': 'error',
+        '@typescript-eslint/ban-tslint-comment': 'error',
+        '@typescript-eslint/class-literal-property-style': 'error',
+        '@typescript-eslint/consistent-indexed-object-style': 'error',
+        '@typescript-eslint/consistent-type-assertions': 'error',
+        '@typescript-eslint/consistent-type-definitions': 'error',
+        '@typescript-eslint/consistent-type-imports': 'error',
+        '@typescript-eslint/explicit-function-return-type': 'error',
+        '@typescript-eslint/explicit-member-accessibility': 'error',
+        '@typescript-eslint/member-delimiter-style': 'error',
+        '@typescript-eslint/member-ordering': 'error',
```

Comment:
[@typescript-eslint/member-ordering](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/member-ordering.md)
This rule groups decorated and abstract members together. This is problematic if not incorrect. If decorated members are grouped it's always because of their type (e.g. Inputs and Outputs). That is not always the case (e.g. HostListener), because decorators are modifiers that don't always imply a relationship with each other or a stronger relationship over other attributes. At the minimum I would group decorated members with their accessibility group members. It might seem reasonable to group public decorated fields together in Angular, because they tend to be Inputs and Outputs. However, that may not make sense with any other decorator or language semantics other than Angular. Therefore, I think it's best not to even group decorated members within accessibility groups, and leave it up to the developer's understanding of their semantics.

As for abstract members, we organize them by their relationship to members within the API as if they aren't abstract and as they would be ordered in documentation. I think they should be handled similarly to decorated members.

This proposal better aligns with our [C# guidelines](https://github.com/ni/csharp-styleguide#c6-%EF%B8%8F-consider-arranging-types-and-members-in-logical-layers) which take an even less restrictive approach in favor of using good judgement to achieve readability. It still organizes by accessibility as the [default](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/member-ordering.md#default-configuration) and the current TSLint rule do. Unfortunately, accessors are [not yet supported](https://github.com/typescript-eslint/typescript-eslint/issues/929), so I included a recommendation for them in the comment.

```typescript
/*
  Group members by fields and methods and then order them by accessibility starting with statics. Order members
  within these groups in a logical organization where readability is the most important thing. Private fields
  that back public accessors may be grouped with their accessor by disabling the rule with a comment.
*/
'@typescript-eslint/member-ordering': ['error', {
  default: [
    'signature',
    'public-static-field',
    'protected-static-field',
    'private-static-field',
    'static-field',
    'public-field',
    'protected-field',
    'private-field',
    'field',
    'public-constructor',
    'protected-constructor',
    'private-constructor',
    'constructor',
    'public-static-method',
    'protected-static-method',
    'private-static-method',
    'static-method',
    'public-method',
    'protected-method',
    'private-method',
    'method'
  ]
}]
```

Reply:
@rajsite 🚀 (unless you vote otherwise)